service/history/workflowExecutionUtil.go

```diff
@@ -25,6 +25,24 @@ import (
 	"github.com/uber/cadence/common"
 )
 
+type workflowLockedContext struct {
```

Comment:
plz name this workflowImpl and create a corresponding interface for it

Reply:
Done